accelerate-llvm-ptx/src/Data/Array/Accelerate/LLVM/PTX/CodeGen/Fold.hs

```diff
@@ -129,11 +129,10 @@ mkFoldAllS dev aenv tp combine mseed marr =
       paramEnv            = envParam aenv
       --
       config              = launchConfig dev (CUDA.incWarp dev) smem multipleOf multipleOfQ
-      smem n              = warps * (1 + per_warp) * bytes
+      smem n              = warps * bytes
         where
           ws        = CUDA.warpSize dev
-          warps     = n `P.quot` ws
-          per_warp  = ws + ws `P.quot` 2
+          warps     = n `P.quot` ws -- why is this rounded down? don't we need a warp for the leftovers? TODO
```

Comment:
n is always a multiple of the warp size; you can check this in D.A.A.LLVM.PTX.Analysis.Launch and Foreign.CUDA.Analysis.Occupancy.